### PR TITLE
fix: add decoratorAutoAccessors plugin

### PR DESCRIPTION
**Pull request by syi0808**
add decoratorAutoAccessors plugin in typescript parser plugins to solve error.

### before screenshot
Screenshot 2024-07-10 at 9 02 52 PM

### after screenshot
Screenshot 2024-07-10 at 9 02 41 PM

fix: #580

### CODE REVIEW

**Comment by Daniel15** [comment]
Can you please provide an example of code that requires this option?

**Comment by ElonVolo** [comment]
Here's an example, @Daniel15 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html

To kill two birds with one stone, I think it would be great if in addition to providing the awesome fix for this bug, syi0808 could also add a unit test(s) that make use of the accessor keyword in some example code and then makes sure the manipulation of it comes out correctly.

That way we can guard against any future regressions with the parsing.

**Comment by syi0808** [comment]
Thanks for reviews! I add test for changes now. But, before this pr e2e test didn't exist. So i add separated test for e2e. Will it be okay?

**Comment by doouding** [comment]
I encountered this problem too. Looking forward to it.

File: test/__tests__/ts-decorator-auto-accessor-test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"use strict";
```
Comment (Daniel15) [inline test/__tests__/ts-decorator-auto-accessor-test.js:1]: Can these go in the existing `src/__tests__` directory?

File: test/__tests__/ts-decorator-auto-accessor-test.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+jest.autoMockOff();
+const defineTest = require("../../src/testUtils").defineTest;
+
+describe("should be parse typescript decoratorAutoAccessors correctly", function () {
+  defineTest(__dirname, "ts-decorator-auto-accessor", null, null, {
```
Comment (Daniel15) [inline test/__tests__/ts-decorator-auto-accessor-test.js:7]: Please use `defineInlineTest`. You can also define the transform inline if you use `defineInlineTest`, e.g.

```
function transformer(file, api) {
  .....
}

defineInlineTest(transformer, .......)
```

**Review by Daniel15 — APPROVED** [review]

**Comment by Daniel15** [comment]
Sorry for the delay in getting back to this. I added some comments inline.

File: test/__tests__/ts-decorator-auto-accessor-test.js (hunk 1 above)
Comment (syi0808) [reply test/__tests__/ts-decorator-auto-accessor-test.js]: Okay. I moved the test.

File: test/__tests__/ts-decorator-auto-accessor-test.js (hunk 2 above)
Comment (syi0808) [reply test/__tests__/ts-decorator-auto-accessor-test.js]: Thanks for review. I switched to inline test.

**Comment by Daniel15** [comment]
Thanks!

**Comment by Daniel15** [comment]
Released in v17.0.0. Thanks for your contribution!

**Pull request merged**